flask_backend/repository/movies.py

```diff
@@ -22,18 +24,23 @@ def get_all(include_drafts: bool = False) -> List[Optional[Movie]]:
 def get_paginated(
     current_page: int, per_page: int, include_drafts: bool = False
 ) -> List[Optional[Movie]]:
-    if current_page == 1:
-        offset_value = 0
-    else:
-        offset_value = current_page * per_page
-
-    query = db_session.query(Movie).join(Screening).order_by(Screening.id.desc())
+    offset_value = (current_page - 1) * per_page
+
+    query = (
+        db_session.query(Screening)
+        .filter(
+            Screening.id.in_(
+                db_session.query(func.max(Screening.id)).group_by(Screening.movie_id)
+            )
+        )
+        .order_by(Screening.id.desc())
+    )
```

Comment:
tchê eu acho que esse `.filter` aqui pode ser removida por dois motivos:

1. Ela deixa o query meio complexo, e talvez conforme a quantidade de filmes aumente, acabe tornando o endpoint mais lento. É interessante que aqui seja o mais rápido possível pq tem impacto na fluidez da página dos posters. (admito que não sei dizer se deixa lento ou não, só tô chutando).
2. Esse filtro só permite uma screening por filme. Acontece que muitos filmes passam em mais de um cinema. Um exemplo disso é o filme "Motel Destino". Ele tem uma screening na Cinemateca Paulo Amorim, com esse poster: <https://i.ibb.co/hcBbc5N/ed26e96a9936.png> e outra screening no CineBancários, com esse poster: <https://i.ibb.co/wKBzhLD/f346e77a6585.webp>.

Eu testei aqui e essa solução, com ou sem o filtro, resolve o problema anterior onde as screenings vinham repetidas pra diferentes valores de paginação (`current_page`).

```suggestion
    query = (
        db_session.query(Screening)
        .order_by(Screening.id.desc())
    )
```

Que tal removermos o filtro?

Reply:
quando eu estava desenvolvendo eu até pensei nesse segundo caso que você comentou sobre o filme "Motel Destino" ter dois posters e essa modificação iria exibir apenas 1 deles, porem na estrutura do projeto atual eu não consegui pensar em nenhum outro jeito, na minha visão eu acho melhor omitir alguns filmes do que continuar repetindo, depois podemos pensar em uma solução mais robusta.

Reply:
Opa, não sei se ficou claro, mas eu removi só a parte do `func.max(Screening.id)` e reparei que a solução seguiu funcionando!